darts/models/forecasting/tsmixer_model.py

```diff
@@ -0,0 +1,943 @@
+        activation: Callable
+            The activation function used in feature mixing.
+        dropout: float
+            The dropout probability used in the feature mixing operation.
+        normalize_before: bool
+            Whether to apply normalization before or after feature mixing.
+        norm_type: type[nn.Module]
+            The type of normalization to use.
+        """
+        super().__init__()
+        self.static_dim = static_dim
+        if self.static_dim != 0:
+            self.fr_static: nn.Linear | None = nn.Linear(static_dim, output_dim)
+            feature_mixing_input = input_dim + output_dim
+        else:
+            self.fr_static = None
+            feature_mixing_input = input_dim
+
+        self.fm = FeatureMixing(
+            sequence_length=sequence_length,
+            input_dim=feature_mixing_input,
+            output_dim=output_dim,
+            ff_dim=ff_dim,
+            activation=activation,
+            dropout=dropout,
+            normalize_before=normalize_before,
+            norm_type=norm_type,
+        )
+
+    def forward(self, x: torch.Tensor, x_static: torch.Tensor | None) -> torch.Tensor:
+        if self.fr_static is None:
+            return self.fm(x)
+        v = self.fr_static(x_static)
+        repeat_factor = x.size(1) // v.size(1)
+        v = v.repeat(1, repeat_factor, 1)
+        return self.fm(torch.cat([x, v], dim=-1))
+
+
+class TimeMixing(nn.Module):
+    def __init__(
+        self,
+        sequence_length: int,
+        input_dim: int,
+        activation: Callable,
+        dropout: float,
+        norm_type: type[nn.Module],
+    ) -> None:
+        """Applies a transformation over the time dimension of a sequence based on the
+        `PyTorch implementation of TSMixer <https://github.com/ditschuk/pytorch-tsmixer>`_.
+
+        This module applies a linear transformation followed by an activation function
+        and dropout over the sequence length of the input feature torch.Tensor after converting
+        feature maps to the time dimension and then back.
+
+        Parameters
+        ----------
+        sequence_length: int
+            The length of the sequences to be transformed.
+        input_dim: int
+            The number of input channels to the module.
+        activation: Callable
+            The activation function to be used after the linear
+            transformation.
+        dropout: float
+            The dropout probability to be used after the activation function.
+        norm_type: type[nn.Module]
+            The type of normalization to use.
+        """
+        super().__init__()
+        self.norm = norm_type((sequence_length, input_dim))
+        self.activation = activation
+        self.dropout = MonteCarloDropout(dropout)
+        self.fc1 = nn.Linear(sequence_length, sequence_length)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x_temp = feature_to_time(x)
```

Comment:
why x_temp and not just x?

Reply:
Because it needs x later again and can not overwrite it.

```
    def forward(self, x: torch.Tensor) -> torch.Tensor:
        x_temp = feature_to_time(x)
        x_temp = self.activation(self.fc1(x_temp))
        x_temp = self.dropout(x_temp)
        x_res = time_to_feature(x_temp)
        x_temp = self.norm(x + x_res) # <--- here
        return x_temp
   ```